### PR TITLE
Add all ssh kwargs to sanitize_kwargs method

### PR DESCRIPTION
**Pull request by Ch3LL**
### What does this PR do?
Adding upon this PR: https://github.com/saltstack/salt/pull/59664  adding all ssh options users can safely pass.

### CODE REVIEW

File: tests/pytests/conftest.py [hunk 1]
```diff
@@ -359,6 +359,20 @@ def bridge_pytest_and_runtests():
...
+    return opts
+
+
+@pytest.fixture
+def master_config():
+    opts = salt.config.DEFAULT_MASTER_OPTS.copy()
+    opts["__role"] = "master"
+    return opts
+
+
```
Comment (s0undt3ch) [inline tests/pytests/conftest.py:375]:
```suggestion
```

File: tests/pytests/unit/client/test_ssh.py [hunk 2]
```diff
@@ -57,3 +62,80 @@ def test_cmd_block_python_version_error(ssh_target):
...
+    roster = pathlib.Path(RUNTIME_VARS.TMP_CONF_DIR) / "roster"
+    client = salt.client.ssh.client.SSHClient(
+        mopts=master_config, disable_custom_roster=True
+    )
+    if in_parser:
+        assert opt_key in ssh_kwargs
+
+    with patch("salt.roster.get_roster_file", MagicMock(return_value=roster)):
+        ssh_obj = client._prep_ssh(**opts)
+        assert ssh_obj.opts.get(opt_key, None) == opt_value
```
Comment (s0undt3ch) [inline tests/pytests/unit/client/test_ssh.py:141]:
```suggestion
def test_ssh_kwargs(temp_salt_master, test_opts):
    """
    test all ssh kwargs are not excluded from kwargs
    when preparing the SSH opts
    """

    ssh_kwargs = salt.utils.parsers.SaltSSHOptionParser().defaults
    opt_key = test_opts[0]
    opt_value = test_opts[1]
    # Is the kwarg in salt.utils.parsers?
    in_parser = test_opts[2]

    opts = {
        "eauth": "auto",
        "username": "test",
        "password": "test",
        "client": "ssh",
        "tgt": "localhost",
        "fun": "test.ping",
        opt_key: opt_value,
    }
    client = salt.client.ssh.client.SSHClient(
        mopts=temp_salt_master.config.copy(), disable_custom_roster=True
    )
    if in_parser:
        assert opt_key in ssh_kwargs

    with patch("salt.roster.get_roster_file", MagicMock(return_value="")):
        ssh_obj = client._prep_ssh(**opts)
        assert ssh_obj.opts.get(opt_key, None) == opt_value
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by s0undt3ch** [comment]
Let's try the above and see if it's enough.

**Comment by Ch3LL** [comment]
re-run full all

**Comment by Ch3LL** [comment]
re-run full all

**Comment by Ch3LL** [comment]
re-run full all

**Review by s0undt3ch — DISMISSED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Review by cmcmarrow — APPROVED** [review]

**Pull request closed without merging**